src/spurt/utils/_tiling.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+
+from typing import Protocol, runtime_checkable
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+def intersects(box1: ArrayLike, box2: ArrayLike) -> bool:
+    """Check if two boxes intersect."""
+    return not (
+        box1[2] < box2[0] or box1[0] > box2[2] or box1[1] > box2[3] or box1[3] < box2[1]
+    )
+
+
+@runtime_checkable
+class TilerInterface(Protocol):
+    """Rectangular tiler interface."""
+
+    @property
+    def ntiles(self) -> int:
+        """Number of tiles."""
+
+    @property
+    def bounds(self) -> tuple[int, int]:
+        """Shape of rectangle being tiled."""
+
+    @property
+    def tiles(self) -> ArrayLike:
+        """2D array of shape (ntiles, 4) using shapely bbox convention."""
```

Comment:
Done. Working with this template in redoing the tiling interface.

Reply:
https://github.com/isce-framework/dolphin/blob/main/src/dolphin/_types.py#L31 this is what I put in Dolphin, which I may have directly taken from `rasterio`. 

If we add that suggestion, I'd probably call that `classmethod` something else, since I would expect `from_shapely` to accept shapely objects like polygons

Reply:
Also the reason to subclass `typing.NamedTuple` is that it lets you unpack bounding boxes nicely, like 
```python
left, bottom, right, top = bbox_instance
```

Reply:
I'll use `from_shapely_bounds` - we can always revisit this later. Would prefer to keep this as ints for now.